vms/platformvm/block/builder/network.go

```diff
@@ -120,22 +121,42 @@ func (n *network) AppGossip(_ context.Context, nodeID ids.NodeID, msgBytes []byt
 	return nil
 }
 
-func (n *network) GossipTx(tx *txs.Tx) error {
+func (n *network) GossipTx(ctx context.Context, tx *txs.Tx) error {
+	txBytes := tx.Bytes()
+	msg := &message.Tx{
+		Tx: txBytes,
+	}
+	msgBytes, err := message.Build(msg)
+	if err != nil {
+		return err
+	}
+
 	txID := tx.ID()
+	n.gossipTx(ctx, txID, msgBytes)
+	return nil
+}
+
+func (n *network) gossipTx(ctx context.Context, txID ids.ID, msgBytes []byte) {
+	// This lock is just to ensure there isn't racy behavior between checking if
+	// the tx was gossiped and marking the tx as gossiped.
```

Comment:
nit: I would probably not document this because the usage of  a lock in the following line makes it pretty clear that it's being used to prevent a racy read/write.

Reply:
I'm down to delete